polaris.shopify.com/pages/examples/index-table-with-nested-rows.tsx

```diff
@@ -0,0 +1,233 @@
+  LegacyCard,
+  Text,
+  useIndexResourceState,
+  IndexTable,
+} from '@shopify/polaris';
+import type {IndexTableRowProps, IndexTableProps} from '@shopify/polaris';
+import React, {Fragment} from 'react';
+import {withPolarisExample} from '../../src/components/PolarisExampleWrapper';
+
+export function WithNestedRowsExample() {
+  interface Product {
+    id: string;
+    quantity: number;
+    price: string;
+    size: string;
+    color: string;
+    image?: string;
+    disabled?: boolean;
+  }
+
+  interface ProductRow extends Product {
+    position: number;
+  }
+
+  interface ProductGroup {
+    id: string;
+    position: number;
+    products: ProductRow[];
+  }
+
+  interface Groups {
+    [key: string]: ProductGroup;
+  }
+
+  const rows: Product[] = [
+    {
+      id: '3411',
+      quantity: 11,
+      price: '$2,400',
+      size: 'Small Small Small',
+      color: 'Orange Orange Orange',
+    },
+    {
+      id: '2562',
+      quantity: 30,
+      price: '$975',
+      size: 'Medium Medium Medium',
+      color: 'Orange Orange Orange',
+    },
+    {
+      id: '4102',
+      quantity: 27,
+      price: '$2885',
+      size: 'Large',
+      color: 'Orange',
+    },
+    {
+      id: '2564',
+      quantity: 19,
+      price: '$1,209',
+      size: 'Small',
+      color: 'Red',
+      disabled: true,
+    },
+    {
+      id: '2563',
+      quantity: 22,
+      price: '$1,400',
+      size: 'Small',
+      color: 'Green',
+    },
+  ];
+
+  const columnHeadings = [
+    {title: 'Name', id: 'column-header--size'},
+    {
+      hidden: false,
+      id: 'column-header--price',
+      title: 'Price',
+    },
+    {
+      alignment: 'end',
+      id: 'column-header--quantity',
+      title: 'Available',
+    },
+  ];
+
+  const groupRowsByGroupKey = (
+    groupKey: keyof Product,
+    resolveId: (groupVal: string) => string,
+  ) => {
+    let position = -1;
+    const groups: Groups = rows.reduce((groups: Groups, product: Product) => {
+      const groupVal: string = product[groupKey] as string;
+      if (!groups[groupVal]) {
+        position += 1;
+
+        groups[groupVal] = {
+          position,
+          products: [],
+          id: resolveId(groupVal),
+        };
+      }
+      groups[groupVal].products.push({
+        ...product,
+        position: position + 1,
+      });
+
+      position += 1;
+      return groups;
+    }, {});
+
+    return groups;
+  };
+
+  const resourceName = {
+    singular: 'product',
+    plural: 'products',
+  };
+
+  const {selectedResources, allResourcesSelected, handleSelectionChange} =
+    useIndexResourceState(rows as unknown as {[key: string]: unknown}[], {
+      resourceFilter: ({disabled}) => !disabled,
+    });
+
+  const groupedProducts = groupRowsByGroupKey(
+    'color',
+    (color) => `color--${color.toLowerCase()}`,
+  );
+
+  const rowMarkup = Object.keys(groupedProducts).map((color, index) => {
+    const {products, position, id: subheaderId} = groupedProducts[color];
```

Comment:
This is not a subheader, let's call it a parentId
```suggestion
    const {products, position, id: parentId} = groupedProducts[color];
```

Reply:
Or `productId`?

Reply:
@kyledurand can this example also be tested somehow the similar way I run stories?

Reply:
You you can build the site locally. Should just be `yarn dev --filter=polaris.shopify.com`

Reply:
Thank you @kyledurand ! For some reasons I'm getting an error and site never opens even on polaris `main` branch. 🤔 
<img width="1079" alt="image" src="https://github.com/Shopify/polaris/assets/104942025/c3880f56-61bb-403c-b191-a111abf0f2f1">

Reply:
resolved in  polaris channel